File: frontend/profiles/lavanderia/lavanderia-order-status.ts
/**
 * Status de um pedido lavanderia (delivery com coleta+entrega, camada 8.10) — espelho 1:1 de
 * src/main/java/com/meada/whatsapp/profiles/lavanderia/orders/LavanderiaOrderStatus.java.
 *
 * O LavanderiaOrderStatusParityTest (backend) garante que os ids aqui e no enum Java nunca divergem
 * (o teste casa textualmente cada objeto `{ id: '...' }` deste arquivo). A CHECK constraint de
 * lavanderia_orders.status trava os mesmos ids no banco.
 *
 * Gate de aceite humano (ação no painel, nunca da IA). Transições:
 *   aguardando   → coletado, recusado, cancelado
 *   coletado     → em_processo, cancelado
 *   em_processo  → pronto, cancelado
 *   pronto       → saiu_entrega, cancelado
 *   saiu_entrega → entregue, cancelado
 *   entregue/recusado/cancelado → (terminal)
 */
export const LAVANDERIA_ORDER_STATUSES = [
  { id: 'aguardando', label: 'Aguardando aceite' },
  { id: 'coletado', label: 'Coletado' },
  { id: 'em_processo', label: 'Em processo' },
  { id: 'pronto', label: 'Pronto' },
  { id: 'saiu_entrega', label: 'Saiu pra entrega' },
  { id: 'entregue', label: 'Entregue' },
  { id: 'recusado', label: 'Recusado' },
  { id: 'cancelado', label: 'Cancelado' },
] as const

export type LavanderiaOrderStatus = (typeof LAVANDERIA_ORDER_STATUSES)[number]
export type LavanderiaOrderStatusId = LavanderiaOrderStatus['id']

export const ALLOWED_NEXT: Record<LavanderiaOrderStatusId, LavanderiaOrderStatusId[]> = {
  aguardando: ['coletado', 'recusado', 'cancelado'],
  coletado: ['em_processo', 'cancelado'],
  em_processo: ['pronto', 'cancelado'],
  pronto: ['saiu_entrega', 'cancelado'],
  saiu_entrega: ['entregue', 'cancelado'],
  entregue: [],
  recusado: [],
  cancelado: [],
}

export function statusLabel(id: string): string {
  return LAVANDERIA_ORDER_STATUSES.find((s) => s.id === id)?.label ?? id
}

File: src/main/java/com/meada/profiles/lavanderia/orders/LavanderiaOrderStatus.java
package com.meada.profiles.lavanderia.orders;

import java.util.Arrays;
import java.util.Optional;
import java.util.Set;

/**
 * Status de um pedido lavanderia (camada 8.10) — clone de
 * {@link com.meada.profiles.floricultura.orders.FloriculturaOrderStatus} adaptado ao fluxo de
 * lavanderia (coleta → processamento → entrega). Inclui o GATE DE ACEITE HUMANO (aguardando →
 * coletado / recusado, ações HUMANAS no painel).
 * <pre>
 *   aguardando    → coletado, recusado, cancelado
 *   coletado      → em_processo, cancelado
 *   em_processo   → pronto, cancelado
 *   pronto        → saiu_entrega, cancelado
 *   saiu_entrega  → entregue, cancelado
 *   entregue      → (terminal)
 *   recusado      → (terminal)
 *   cancelado     → (terminal)
 * </pre>
 * Transição inválida → 409 invalid_status_transition no controller. Espelhado 1:1 por
 * {@code frontend/profiles/lavanderia/lavanderia-order-status.ts} (LavanderiaOrderStatusParityTest).
 */
public enum LavanderiaOrderStatus {
    AGUARDANDO("aguardando"),
    COLETADO("coletado"),
    EM_PROCESSO("em_processo"),
    PRONTO("pronto"),
    SAIU_ENTREGA("saiu_entrega"),
    ENTREGUE("entregue"),
    RECUSADO("recusado"),
    CANCELADO("cancelado");

    private final String id;

    LavanderiaOrderStatus(String id) {
        this.id = id;
    }

    public String id() {
        return id;
    }

    public static Optional<LavanderiaOrderStatus> fromId(String id) {
        if (id == null) {
            return Optional.empty();
        }
        return Arrays.stream(values()).filter(s -> s.id.equals(id)).findFirst();
    }

    /** Transições permitidas a partir deste status. */
    public Set<LavanderiaOrderStatus> allowedNext() {
        return switch (this) {
            case AGUARDANDO -> Set.of(COLETADO, RECUSADO, CANCELADO);
            case COLETADO -> Set.of(EM_PROCESSO, CANCELADO);
            case EM_PROCESSO -> Set.of(PRONTO, CANCELADO);
            case PRONTO -> Set.of(SAIU_ENTREGA, CANCELADO);
            case SAIU_ENTREGA -> Set.of(ENTREGUE, CANCELADO);
            case ENTREGUE, RECUSADO, CANCELADO -> Set.of();
        };
    }

    public boolean canTransitionTo(LavanderiaOrderStatus next) {
        return allowedNext().contains(next);
    }

    /**
     * Texto fixo da notificação outbound disparada ao ENTRAR neste status. null = não notifica
     * ({@code aguardando} é silencioso — a IA já confirmou o recebimento na própria mensagem). No caso
     * de {@code recusado}, o MOTIVO (rejection_reason) é concatenado pelo Service, não aqui.
     */
    public String notificationText() {
        return switch (this) {
            case COLETADO -> "Recebemos suas peças, já vamos cuidar delas com todo o carinho. 🧺";
            case PRONTO -> "Boa notícia! Suas peças estão prontas. Em breve enviamos para entrega.";
            case SAIU_ENTREGA -> "Suas peças saíram para entrega. Logo chegam aí!";
            case ENTREGUE -> "Pedido entregue! Esperamos que esteja tudo impecável. ✨";
            case RECUSADO -> "Infelizmente não conseguimos atender seu pedido. Pedimos desculpa pelo transtorno.";
            case CANCELADO -> "Seu pedido foi cancelado. Se quiser refazer, é só me chamar.";
            // aguardando/em_processo não notificam (recebimento já confirmado; processamento é interno).
            case AGUARDANDO, EM_PROCESSO -> null;
        };
    }
}
